dali/test/python/test_utils.py

```diff
@@ -93,15 +108,26 @@ def is_error(mean_err, max_err, eps, max_allowed_error):
         batch1 = batch1.as_cpu()
     if isinstance(batch2, dali.backend_impl.TensorListGPU):
         batch2 = batch2.as_cpu()
+    # Check layouts where possible
+    for batch in [batch1, batch2]:
+        if expected_layout is not None and isinstance(batch, dali.backend.TensorListCPU):
+            assert batch.layout() == expected_layout, \
+                'Unexpected layout, expected "{}", got "{}".'.format(expected_layout, batch.layout())
+
+    if compare_layouts and \
```

Comment:
isn't the first block of code enough? Why are you checking it again?

Reply:
It's to compare between two batches (as opposed to providing the layout as argument to this function and checking if (where it can be check) we have a match with expected layout.